components/sponsor/NftSubmissionForm.vue

```diff
@@ -0,0 +1,552 @@
+<script setup lang="ts">
+import type { NftSubmission } from '~/types';
+import { useVuelidate } from '@vuelidate/core';
+import { helpers, maxLength, minLength, numeric, required } from '@vuelidate/validators';
+import { get, set } from '@vueuse/shared';
+import { useRotkiSponsorshipPayment } from '~/composables/rotki-sponsorship/payment';
+import { toMessages } from '~/utils/validation';
+
+interface Props {
+  address: string | undefined;
+  isConnected: boolean;
+  editingSubmission?: NftSubmission;
+}
+
+const props = defineProps<Props>();
+
+const emit = defineEmits<{
+  'submission-success': [];
+  'cancel-edit': [];
+}>();
+
+const { t } = useI18n();
+const { fetchNftMetadata } = useNftMetadata();
+
+const displayName = ref<string>('');
+const signature = ref<string>('');
+const signatureTimestamp = ref<number>(0);
+const hasSigned = ref<boolean>(false);
+const isSigning = ref<boolean>(false);
+const isSubmitting = ref<boolean>(false);
+const error = ref<string>('');
+const success = ref<boolean>(false);
+const imageFile = ref<File | null>(null);
+const imagePreview = ref<string>('');
+const tokenId = ref<string>('');
+const email = ref<string>('');
+const isCheckingNft = ref<boolean>(false);
+const nftTier = ref<'bronze' | 'silver' | 'gold' | undefined>();
```

Comment:
don't we have a typealias for the Nft tiers? I see them repeated in a lot of places.

Also isn't undefined redundant here as it is implied since we don't initialize the value?